src/MSXMotherBoard.cc

```diff
@@ -1018,36 +1037,55 @@ void RemoveExtCmd::tabCompletion(std::vector<string>& tokens) const
 StoreSetupCmd::StoreSetupCmd(MSXMotherBoard& motherBoard_)
 	: Command(motherBoard_.getCommandController(), "store_setup")
 	, motherBoard(motherBoard_)
+	, depthOptionMap({
+		{ "none", MSXMotherBoard::SetupDepth::NONE },
+		{ "machine_only", MSXMotherBoard::SetupDepth::MACHINE_ONLY },
+		{ "with_extensions", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS },
+		{ "with_extensions_and_pluggables", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS_AND_PLUGGABLES },
+		{ "with_extensions_and_pluggables_and_media", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS_AND_PLUGGABLES_AND_MEDIA },
+		{ "complete_state", MSXMotherBoard::SetupDepth::COMPLETE_STATE }
+	})
 {
 }
 
 void StoreSetupCmd::execute(std::span<const TclObject> tokens, TclObject& result)
 {
-	checkNumArgs(tokens, Between{1, 2}, Prefix{1}, "?filename?");
+	checkNumArgs(tokens, Between{2, 3}, Prefix{1}, "depth ?filename?");
+
+	std::string depthArg = std::string(tokens[1].getString());
+	if (depthOptionMap.count(depthArg) == 0) {
+		throw CommandException("Unknown depth argument: ", depthArg);
+	}
+
+	if (depthOptionMap.at(depthArg) == MSXMotherBoard::SetupDepth::NONE) return;
 
 	std::string_view filenameArg;
-	if (tokens.size() == 2) {
-		filenameArg = tokens[1].getString();
+	if (tokens.size() == 3) {
+		filenameArg = tokens[2].getString();
 	}
 
 	auto filename = FileOperations::parseCommandFileArgument(
 		filenameArg, Reactor::SETUP_DIR, motherBoard.getMachineName(), Reactor::SETUP_EXTENSION);
 
-	// TODO: make level and parts of levels to be saved configurable (via command arguments?)
-	motherBoard.storeAsSetup(filename);
+	// TODO: make parts of levels to be saved configurable?
```

Comment:
What do you mean with this comment "part of levels"? Something like only some of the extensions/pluggables/...? I don't think we should do that, too complex for little benefit. E.g. too complex error checking (E.g. when is it allowed to omit which extension ... not when that extension makes a pluggable or media slot available that is still part of the setup).

Reply:
It would be possible in an easier way if we only allow it only at the 'leaf' depth/level. But anyway, it's a luxury feature that is optional.